project/src/com/google/daggerqueryui/scripts/binding_graph.js

```diff
@@ -138,6 +167,50 @@ const bindingGraph = (function() {
     }).length !== 0;
   }
 
+  /**
+   * Supports click and double click events in the network object.
+   * @param {vis.Network} network
+   */
+  function supportEventsRecognition(network) {
+    // An event for managing children nodes.
+    network.on("doubleClick", function (params) {
+      // Checks if any node was selected.
+      if (params.nodes.length === 0) {
+        return;
+      }
+
+      const nodeId = params.nodes[0];
+      const nodeTitle = nodes.get(nodeId).title;
+
+      // If the dependencies are not currently shown, we prefer to draw them all.
+      // Otherwise, we hide all children, even if they are not all were presented.
```

Comment:
Thanks for the explanation! Would it be worth extracting the logic to a helper method, e.g. `showOrHideDeps`?

Reply:
Yes, good idea. I [have also moved](https://github.com/googleinterns/dagger-query/pull/47/commits/9faa992a1507a10c9bc6120f0c33a46dcdcae425) these comments to the JSDoc.